internal/lms/library/service_test.go

```diff
@@ -0,0 +1,126 @@
+package lms
+
+import (
+	"Demo/config"
+	"Demo/internal/entities/library"
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/gin-gonic/gin"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestStoreBook(t *testing.T) {
+	router := gin.Default()
+
+	db := config.LoadConfigTest()
+	db.Exec("TRUNCATE TABLE libraries;")
+
+	router.POST("/libraries", StoreBook(db))
+
+	bookPayload := library.Library{
+		Book_ID: 1,
+		Aisle: 1,
+		Level: 2,
+		Position: 3,
+	}
+
+	payloadJSON, _ := json.Marshal(bookPayload)
+
+	req, _ := http.NewRequest("POST", "/libraries", bytes.NewBuffer(payloadJSON))
+	req.Header.Set("Content-Type", "application/json")
+
+	resp := httptest.NewRecorder()
+	router.ServeHTTP(resp, req)
+
+	if resp.Code != http.StatusCreated {
+		t.Errorf("Expected status code %d but got %d", http.StatusCreated, resp.Code)
+	}
+
+	var createdBook library.Library
+	err := json.Unmarshal(resp.Body.Bytes(), &createdBook)
+	if err != nil {
+		t.Errorf("Error parsing response body: %v", err)
+	}
+
+	assert.Equal(t, bookPayload.Book_ID, createdBook.Book_ID)
+	assert.Equal(t, bookPayload.Aisle, createdBook.Aisle)
+	assert.Equal(t, bookPayload.Level, createdBook.Level)
+	assert.Equal(t, bookPayload.Position, createdBook.Position)
```

Comment:
please modify it to compare all the properties of a struct in one go. refer to the disputes service as an example.

Reply:
we should not added assert for each entity separately.

Reply:
https://github.com/razorpay/disputes/blob/7bb7df690effd7ad8820a161a477b2291634852a/tests/functional/dispute_fetch_tests/fetchAdminTestData.go#L31

check how we have created test data here. try to see if we can follow this practice.

Reply:
I am able to compare the struct in one go in every test except the create entry test for all tables.